generators/base-application/generator-ts.mts

```diff
@@ -0,0 +1,51 @@
+import _ from 'lodash';
+import type Storage from 'yeoman-generator/lib/util/storage.js';
+
+import BaseGenerator from '../base/index.mjs';
+import { JHIPSTER_CONFIG_DIR } from '../generator-constants.mjs';
+import { getEntitiesFromDir } from './support/index.mjs';
+
+const { upperFirst } = _;
+
+// Temporary Generator with Typescript implementations
+export default class BaseApplicationTsGenerator extends BaseGenerator {
+  /**
+   * Get all the generator configuration from the .yo-rc.json file
+   * @param entityName - Name of the entity to load.
+   * @param create - Create storage if doesn't exists.
+   */
+  getEntityConfig(entityName: string, create = false): Storage | undefined {
+    const entityPath = this.destinationPath(JHIPSTER_CONFIG_DIR, `${upperFirst(entityName)}.json`);
+    if (!create && !this.fs.exists(entityPath)) return undefined;
+    return this.createStorage(entityPath, { sorted: true } as any);
+  }
+
+  /**
+   * get sorted list of entitiy names according to changelog date (i.e. the order in which they were added)
+   */
+  getExistingEntityNames(): string[] {
```

Comment:
Should be in a support file isn't it?

Reply:
IMO no.
Every core api should be implemented inline in the generator (except some utilities).
And we should avoid utilities to use any complex parameter like the generator and the log.

Reply:
Why am I stating that: these APIs will certainly be used by many generators, so we'll avoid passing complex parameters (`generator`) reusing them by moving this method to a dedicated `support` package.

I agree `generator`should not be passed as an attribute. LogAdapter looks to be a must (quite systematic) as there's no slf4j-like static injection fwk